platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditor.java

```diff
@@ -266,6 +280,96 @@ public void addTrustedCertificate(String alias, String keyPassword, String store
     }
 
     @Override
+    public List<Map<String, Object>> addTrustedCertificateFromUrl(String url) {
+        SSLSocket socket = null;
+        String decodedUrl = null;
+        List<Map<String, Object>> resultList = new ArrayList<>();
+        try {
+            decodedUrl = new String(Base64.getDecoder()
+                    .decode(url), "UTF-8");
+            socket = getNonVerifyingSslSocket(decodedUrl);
+            socket.startHandshake();
+            X509Certificate[] peerCertificateChain = (X509Certificate[]) socket.getSession()
+                    .getPeerCertificates();
+            for (X509Certificate certificate : peerCertificateChain) {
+                try {
+                    X500Name x500name = new JcaX509CertificateHolder(certificate).getSubject();
+                    RDN cn = x500name.getRDNs(BCStyle.CN)[0];
+                    String cnStr = IETFUtils.valueToString(cn.getFirst()
+                            .getValue());
+                    trustStore.setCertificateEntry(cnStr, certificate);
+                    resultList.add(Collections.singletonMap("success", true));
```

Comment:
So if there were three certificates, two had encoding issues, and one worked okay, the result list would look like `[["success": false], ["success": false], ["success": true]]`?
That isn't returning much context.